File: src/codes/Users_Manual/geometry_4.py
~from pyautd3 import Controller, AUTD3
~from pyautd3.link.audit import Audit
~autd: Controller[Audit] = Controller.builder().add_device(AUTD3([0.0, 0.0, 0.0])).open(Audit.builder())
dev = autd.geometry[0]
for dev in autd.geometry:
    pass

File: src/codes/Users_Manual/geometry_4.rs
# use autd3::prelude::*;
# 
# #[allow(unused_variables)]
# #[tokio::main]
# async fn main() -> Result<(), Box<dyn std::error::Error>> {
# let autd = Controller::builder([AUTD3::new(Vector3::zeros())])
#    .open(autd3::link::Nop::builder()).await?;
let dev = &autd.geometry[0];
for dev in &autd.geometry {
    // do something
}
# Ok(())
# }